src/geometry/manifold/tangent_bundle.lean

```diff
@@ -455,28 +447,30 @@ def tangent_bundle_core : basic_smooth_bundle_core I M E :=
         (is_open.mem_nhds (local_homeomorph.open_source _) _),
       simpa only [model_with_corners.left_inv] using hx },
     rw [B, C, D, E] at A,
-    simp only [A, continuous_linear_map.coe_comp'] with mfld_simps
+    simp only [A, continuous_linear_map.coe_comp'] with mfld_simps,
+    simp only [forall_const, eq_self_iff_true, linear_map.coe_comp, continuous_linear_map.coe_comp],
   end }
 
 variable {M}
 include I
 
 /-- The tangent space at a point of the manifold `M`. It is just `E`. We could use instead
-`(tangent_bundle_core I M).to_topological_fiber_bundle_core.fiber x`, but we use `E` to help the
+`(tangent_bundle_core I M).to_topological_vector_bundle_core.fiber x`, but we use `E` to help the
 kernel.
 -/
-@[nolint unused_arguments]
-def tangent_space (x : M) : Type* := E
+@[nolint unused_arguments, reducible]
```

Comment:
Please don't make it reducible. We want to hide a little bit from the system (at least from typeclass inference) that this is a complicated object, for better performance.

Reply:
Ok, to be fair I don't remember why I did it

Reply:
Ok the reason why the last build failed is why I had to make it reducible. Lean does not find instances of stuff otherwise and things do not type check. To be fair I do not notice any change in performance when making it reducible: things typecheck in more or less the same amount of time and I don't understand why that extra level should be heavy computationally. If you want to keep it reducible can you have a try yourself? I don't know what is going on type-class-inference wise, but to me it would make sense that things should be reducible up to the vector bundle core, since we are using lemmas for that.

Reply:
You should do as in done currently in master: don't make the definition reducible, but in a section make it reducible and register all the instances you care about.